sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/cloud/service/implementation/servicebus/factory/AbstractServiceBusSubClientBuilderFactory.java

```diff
@@ -64,40 +76,71 @@ protected boolean isShareServiceBusClientBuilder() {
         return shareServiceBusClientBuilder;
     }
 
+    public void addClientBuilderCustomizer(AzureServiceClientBuilderCustomizer<ServiceBusClientBuilder> clientBuilderCustomizer) {
+        if (!isShareServiceBusClientBuilder()) {
+            this.clientBuilderCustomizers.add(clientBuilderCustomizer);
+        }
+    }
+
     @Override
-    protected BiConsumer<T, ProxyOptions> consumeProxyOptions() {
-        return (builder, proxy) -> {
-            if (!isShareServiceBusClientBuilder()) {
-                this.serviceBusClientBuilder.proxyOptions(proxy);
+    public void setSpringIdentifier(String springIdentifier) {
+        if (!isShareServiceBusClientBuilder()) {
+            if (!StringUtils.hasText(springIdentifier)) {
+                LOGGER.warn("SpringIdentifier is null or empty, skip for non-shared ServiceBusClientBuilderFactory.");
+                return;
             }
-        };
+
+            this.springIdentifier = springIdentifier;
+        }
     }
 
     @Override
-    protected BiConsumer<T, AmqpTransportType> consumeAmqpTransportType() {
-        return (builder, t) -> {
-            if (!isShareServiceBusClientBuilder()) {
-                this.serviceBusClientBuilder.transportType(t);
+    public void setDefaultTokenCredential(TokenCredential defaultTokenCredential) {
+        if (!isShareServiceBusClientBuilder()) {
+            if (defaultTokenCredential != null) {
+                this.defaultTokenCredential = defaultTokenCredential;
+            } else {
+                LOGGER.debug("Will ignore the 'null' default token credential, skip for non-shared ServiceBusClientBuilderFactory.");
             }
-        };
+        }
     }
 
     @Override
-    protected BiConsumer<T, AmqpRetryOptions> consumeAmqpRetryOptions() {
-        return (builder, retry) -> {
-            if (!isShareServiceBusClientBuilder()) {
-                this.serviceBusClientBuilder.retryOptions(retry);
+    public void setConnectionStringProvider(ServiceConnectionStringProvider<?> connectionStringProvider) {
```

Comment:
This class should override the `consumeXXX` ops, instead of adding the set methods. Otherwise, it can't use the construction template defined in the base class.